web/server/codechecker_server/api/report_server.py

```diff
@@ -211,9 +211,7 @@ def process_report_filter(session, report_filter):
         AND.append(or_(*OR))
 
     if report_filter.reportHash:
-        OR = [Report.bug_id.ilike(conv(rh))
-              for rh in report_filter.reportHash]
-        AND.append(or_(*OR))
+        AND.append(Report.bug_id.in_(report_filter.reportHash))
```

Comment:
@bruntib @gyorb Maybe we can extend the Thrift API ReportFilter data structure with a new field (e.g.: reportHashRgx) to search for bug ids by using regex expressions too, so on the UI we would be able to search not only full size hashes but substrings too. In case of diff we would use the exact match filter field. So reportHash field would do exact match and reportHashRgx would do regex match. What do you think?

Reply:
The full hash string is not printed on the UI right? It can be problematic to get the full hash only from the web UI.
Keeping the old behavior in some cases can be useful (Maybe in the CLI filters too?).

Reply:
What use case would a regex solve in terms of a bug hash (or any hash)? In case somebody is looking for a hash, then the whole hash is in his hands. Ok, sometimes it is more comfortable if the tool provides prefix search for a hash (for example git commands accept commit hash prefixes or docker commands accept image and container hash prefixes). But couldn't we delay this development until an explicit user request arrives?

Reply:
For example in the GUI I do not see the whole hash by default. It will show only a substring of it (~10 characters):

![image](https://user-images.githubusercontent.com/6695818/84117619-1673dc80-aa32-11ea-9562-b80770c02d67.png)

I have to increase the column size to get the full hash.

For me this patch contains a backward compatibility change. We don't know how many user relies on this feature and how many users use this without a full hash. Maybe we should think about it to keep the old behavior as @gyorb said.

I am thinking on the following solution: if the first item in the `report_filter.reportHash` contains a `*` quantifier we will work as before (query with LIKE) otherwise we will query with IN. What do you think @bruntib?

Reply:
This is a good idea, I like it.

Reply:
I think the solution @csordasmarton mentioned should be ok @bruntib where it is checked if the hash starts with (maybe contains) a `*` or not.